packages/joy-roles/src/tabs/Opportunity.controller.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+
+import { Controller, memoize, View } from '@polkadot/joy-utils/index'
+
+import { ITransport } from '../transport'
+
+import {
+  WorkingGroupOpening,
+  OpeningError,
+  OpeningView,
+} from './Opportunities'
+
+type State = {
+  blockTime?: number,
+  opportunity?: WorkingGroupOpening,
+}
+
+export class OpportunityController extends Controller<State, ITransport> {
+  constructor(transport: ITransport, initialState: State = {}) {
+    super(transport, initialState)
+    this.getBlocktime()
+  }
+
+  @memoize()
+  async getOpportunity(id: string | undefined) {
+    if (typeof id === "undefined") {
+      return this.onError("ApplyController: no ID provided in params")
```

Comment:
I guess it's a typo?
```suggestion
      return this.onError("OpportunityController: no ID provided in params")
```